### PR TITLE
Add bindings for sending keyboard keys to OS, reimplement parsing of keyboard input

### LINKED ISSUE #138: Add user defined bindings for keyboard key presses
As requested [here](https://old.reddit.com/r/Vive/comments/b5s0no/advanced_settings_300_steamvr_input_and/ejg0xjy/).

There should be one or more bindings that correspond to a key press or key sequence that is easily user definable.

### PR DESCRIPTION
**Pull request by username223**
Adds SteamVR bindings for sending keys to the OS while in VR. The specification for the input parser can be found in the `docs/specs/` directory. A guide for using the bindings can be found in `docs/`.

This also adds Linux support for parsing inputs, which previously did not work.

There are Qt Unittests for the parser, although they are not exhaustive and will not be run on CI.

Fixes #138.

The keyboard shortcuts were placed in the `Misc` category of actions.

Media keys have also been moved away from `keyboard_input` in order to prepare for a separate Linux implementation that doesn't just send a virtual keystroke.

### CODE REVIEW

File: src/keyboard_input/input_parser.cpp [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+std::optional<Token> getNumberOrLetter( const char& character ) noexcept
+{
+    if ( islower( character ) || isdigit( character ) )
+    {
+        return static_cast<Token>( character );
+    }
+    return std::nullopt;
+}
+
+std::optional<Token> getFunctionNumber( const char& character ) noexcept
```
Comment (ykeara) [inline src/keyboard_input/input_parser.cpp]: This should Be numbers not function Keys?

File: src/keyboard_input/input_parser.cpp [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+        return true;
+    case Token::KEY_z:
+        return true;
+
+    default:
+        return false;
+    }
+}
+
+bool isLiteral( const Token token ) noexcept
```
Comment (ykeara) [inline src/keyboard_input/input_parser.cpp]: or is this supposed to be numbers and not Function Keys?

File: Readme.md [hunk 3]
```diff
@@ -334,12 +335,10 @@ Allows users to temporarily move and rotate the center of the playspace. This al
 
 <img src="docs/screenshots/UtilitiesPage.png" width="600" alt="Utilities Page">
 
-- **Keyboard Utilities:** Some applications (mostly 2D-monitor applications with tackled-on/modded-in VR support) do not support the VR keyboard but require an actual physical keyboard for text input. This utilities allow to send emulated key strokes from VR keyboard input to such application to overcome this limitation. Does not work in desktop mode.
+- **Keyboard Utilities:** Some applications (mostly 2D-monitor applications with tackled-on/modded-in VR support) do not support the VR keyboard but require an actual physical keyboard for text input. This utilities allow to send emulated key strokes from VR keyboard input to such application to overcome this limitation. Does not work in desktop mode. The "Keyboard Input" button works as described [here](docs/specs/Keyboard_Manager-Parser_Spec.txt).
```
Comment (ykeara) [inline Readme.md]: `(docs/specs/Keyboard_Manager-Parser_Spec.txt)` should be .md

File: src/keyboard_input/input_parser.cpp (hunk 1 above)
Comment (username223) [reply src/keyboard_input/input_parser.cpp]: This is called when the preceding character is a capital F (`input_parser.cpp 183-202`), so it's for function keys.

File: src/keyboard_input/input_parser.cpp (hunk 2 above)
Comment (username223) [reply src/keyboard_input/input_parser.cpp]: `isLiteral` is for all symbols.

Comment (ykeara) [reply src/keyboard_input/input_parser.cpp]: k yep mix-up on my end.

**Pull request merged**